packages/bits/src/lib/menu/menu-key-control.service.ts

```diff
@@ -91,27 +102,35 @@ export class MenuKeyControlService implements OnDestroy {
     }
 
     private handleOpenKeyDown(event: KeyboardEvent): void {
-        if (event.keyCode === DOWN_ARROW || event.keyCode === UP_ARROW) {
+        if (event.code === KEYBOARD_CODE.ARROW_DOWN || event.code === KEYBOARD_CODE.ARROW_UP) {
             // passing the event to key manager so we get a change fired
             this.keyboardEventsManager.onKeydown(event);
+            this.announceCurrentItem();
         }
-        if (event.keyCode === PAGE_UP) {
+        if (event.code === KEYBOARD_CODE.PAGE_UP || event.code === KEYBOARD_CODE.HOME  || (event.metaKey && event.code === KEYBOARD_CODE.ARROW_UP)) {
             event.preventDefault();
             this.keyboardEventsManager.onKeydown(event);
             this.keyboardEventsManager.setFirstItemActive();
+            this.announceCurrentItem();
         }
-        if (event.keyCode === PAGE_DOWN) {
+        if (event.code === KEYBOARD_CODE.PAGE_DOWN || event.code === KEYBOARD_CODE.END  || (event.metaKey && event.code === KEYBOARD_CODE.ARROW_DOWN)) {
```

Comment:
I wonder why PAGE_UP and PAGE_DOWN take the user all the way to the top or bottom instead of going page by page. I know this is pre-existing behavior, but maybe you know why?

Reply:
I think if it is here it was discussed with UX. I doubt there are any docs for that though. According to the w3c requirements, PAGE UP and PAGE DOWN keys are not required to participate in the navigation flow^ however, if it is here, it was discussed. I can also tell there was no such a requirement to navigate page by page for the component.

Reply:
okay